helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java

```diff
@@ -106,4 +131,42 @@ protected TicketResponse buildResponse(Ticket ticket, SymUser agent, String mess
 
   protected abstract TicketResponse execute(Ticket ticket, SymUser agentUser);
 
+  protected void sendMessageWithShowHistoryFalse(Ticket ticket, Long agentId) {
+    if (Boolean.FALSE.equals(ticket.getShowHistory())) {
+      SymStream clientStream = new SymStream();
+      clientStream.setStreamId(ticket.getClientStreamId());
+
+      SymStream serviceStream = new SymStream();
+      serviceStream.setStreamId(ticket.getServiceStreamId());
+
+      try {
+        List<SymMessage> messages = symphonyClientUtil.getSymMessages(clientStream,
+            ticket.getQuestionTimestamp(), 100);
+
+        messages.stream()
+            .filter(symMessage -> ticket.getClient()
+                .getUserId()
+                .equals(symMessage.getSymUser().getId()))
+            .sorted(Comparator.comparing(SymMessage::getTimestamp))
+            .forEach(symMessage -> {
+              symMessage.setStream(serviceStream);
```

Comment:
Please use the 'map' method to modify the message object.

```
.sorted(Comparator.comparing(SymMessage::getTimestamp))
.map(symMessage -> {
  symMessage.setStream(serviceStream);
  symMessage.setStreamId(serviceStream.getStreamId());
  return symMessage;
})
.forEach(symMessage -> {
  sendMessage(symMessage, agentId);
});
```

Reply:
Done.